### PR TITLE
Add the K-LMS scheduler to the inpainting pipeline + tests

### LINKED ISSUE #272: Support K-LMS for inpainting
You have added K-LMS support to [image_to_image.py](https://github.com/huggingface/diffusers/blob/main/examples/inference/image_to_image.py), can you add it to [inpainting.py](https://github.com/huggingface/diffusers/blob/main/examples/inference/inpainting.py)?

### LINKED ISSUE #341: K-LMS scheduler for in-paint
**Is your feature request related to a problem? Please describe.**

The K-LMS scheduler currently doesn't work for Stable Diffusion In-painting.

It would be nice to make it work as also asked for in #329 

cc @anton-l 

**Additional context**

There seem to be a couple of issues currently with in-paintinig -> @patil-suraj @anton-l could you take a look here on Monday?

- https://github.com/huggingface/diffusers/pull/328
- https://github.com/huggingface/diffusers/issues/321
- https://github.com/huggingface/diffusers/issues/261

### LINKED ISSUE #562: StableDiffusionInpaintPipeline does not work with LMSDiscreteScheduler
### Describe the bug

`StableDiffusionInpaintPipeline` does not work with `LMSDiscreteScheduler`

### Reason
- This seems to be because of the absence of special operations for `LMSDiscreteScheduler` as performed in `StableDiffusionImg2ImgPipeline` 

https://github.com/huggingface/diffusers/blob/429dace10a356a776f935fc11e16d5b321b496f3/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py#L206-L212
 
https://github.com/huggingface/diffusers/blob/429dace10a356a776f935fc11e16d5b321b496f3/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py#L245

### Proposal:

I think it is better to unify pipelines  as proposed in #551, since most processes are the same.

### Reproduction

_No response_

### Logs

```shell
"/path/to/lib/python3.10/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py", line 245, in __call__
    timesteps = torch.tensor([timesteps] * batch_size, dtype=torch.long, device=self.device)

TypeError: only integer tensors of a single element can be converted to an index
```

### System Info

- diffusers 429dace10a356a776f935fc11e16d5b321b496f3

### LINKED ISSUE #582: LMSDiscreteScheduler support for Inpaint Pipeline
Currently there is no support for the LMSDiscreteScheduler in the StableDiffusionInpaintPipeline

### PR DESCRIPTION
**Pull request by anton-l**
It's been a long time :) 
The changes are mostly ported from the img2img pipeline.

Closes #341
Closes #272
Closes #562
Closes #582

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py [hunk 1]
```diff
@@ -299,15 +311,20 @@ def __call__(
...
+                init_latents_proper = self.scheduler.add_noise(init_latents_orig, noise, t)
 
-            # masking
-            init_latents_proper = self.scheduler.add_noise(init_latents_orig, noise, t)
             latents = (init_latents_proper * mask) + (latents * (1 - mask))
 
         # scale and decode the image latents with vae
         latents = 1 / 0.18215 * latents
-        image = self.vae.decode(latents).sample
+        image = self.vae.decode(latents.to(self.vae.dtype)).sample
```
Comment (patil-suraj) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:327]: why is the required, all models will be in same `dtype` in the pipeline no ?

**Review by patil-suraj — APPROVED** [review]
Very cool!

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py (hunk 1 above)
Comment (anton-l) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py]: Weird, these were previously needed in the img2img pipeline to make it work under `autocast` but that's no longer the case!

Removed all extra type casts from both pipelines now, and checked with and without autocast: everything works :)

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py [hunk 2]
```diff
@@ -265,8 +265,6 @@ def __call__(
                 sigma = self.scheduler.sigmas[t_index]
                 # the model input needs to be scaled to match the continuous ODE formulation in K-LMS
                 latent_model_input = latent_model_input / ((sigma**2 + 1) ** 0.5)
-                latent_model_input = latent_model_input.to(self.unet.dtype)
-                t = t.to(self.unet.dtype)
```
Comment (anton-l) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:269]: These extra type casts are no longer needed, even under `autocast`. Not sure what was causing these problems before, so I'll keep an eye on the issues.

**Pull request merged**